check-oracle.yml

```diff
@@ -0,0 +1,246 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+# check-oracle.yml
+- name: "ORAchk: uninstall, install or run"
+  hosts: all
+  become: true
+  gather_facts: false
+  roles:
+    - role: check-oracle
+
+  tasks:
+
+    - name: Check for AHF Installation
+      stat:
+        path: "{{ AHFCTL_PATH }}"
+      register: ahfctl_script
+
+    - name: Uninstall AHF
+      shell: yes 'Y' | "{{ AHFCTL_PATH }}" uninstall
+      when: (uninstall_ahf | bool) and ahfctl_script.stat.exists
+      ignore_errors: true
+      failed_when: ahf_uninstall_result.rc != 0 and ahf_uninstall_result.rc is defined
+      register: ahf_uninstall_result
+
+    - block:
+        - name: Create AHF directory
+          file:
+            path: "{{ ahf_extract_path }}"
+            state: directory
+            mode: "0700"
+
+        - name: Create ORAchk directory
+          file:
+            path: "{{ orachk_script_dir }}"
+            state: directory
+            mode: "0700"
+
+        - name: Copy AHF file from Google Storage
+          vars:
+            local_ahf_path: "/tmp/{{ AHF_LOCATION | basename }}"
+          block:
+
+            - name: Download AHF file from Google Storage to the ansible control node.
+              local_action: 
+                module: command
+                cmd: "gsutil cp {{ AHF_LOCATION }} {{ local_ahf_path }}"
```

Comment:
Separately, can we fix the quoting here?  I suggest:
cmd: gsutil cp "{{ AHF_LOCATION }}" "{{ local_ahf_path }}"
(or, even better, the equivalent gcloud storage cp command)
Among other things, it will help us detect if one of these variables is empty.

Reply:
Now using  gcloud storage cp "{{ AHF_LOCATION }}" "{{ local_ahf_path }}"